cmd/swagger/setup.go

```diff
@@ -179,3 +183,30 @@ func AccessManager(api *operations.BeAPI, bindings []config.RoleEndpointBinding)
 	}
 	return manager, nil
 }
+
+func runUnblockPeriodically(ctx context.Context, client *ent.Client, checkPeriodDuration time.Duration, lg *zap.Logger) {
```

Comment:
Is the main idea here to stop the job via cancelling its context, rather than use the timer.Stop()?

Is this more or less what you mean
`
func runJobPeriodically(ctx context.Context, checkPeriodDuration time.Duration, lg *zap.Logger, job func(ctx context.Context)) {
    pt := timer.NewPeriodicTimer()
    f := func() {
        select {
        case <-ctx.Done(): 
            lg.Info("Job has been cancelled.")
            return
        default:
            job(ctx)
        }
    }
    pt.Start(checkPeriodDuration, f)
    f() 
}

func runUnblockPeriodically(ctx context.Context, client *ent.Client, checkPeriodDuration time.Duration, lg *zap.Logger) {
    runJobPeriodically(ctx, checkPeriodDuration, lg, func(ctx context.Context) {
        eqRepo := repositories.NewEquipmentRepository()
        numDeleted, err := eqRepo.UnblockAllExpiredEquipment(ctx, client)
        if err != nil {
            lg.Error("error when performing UnblockAllExpiredEquipment()", zap.Error(err))
        } else {
            if numDeleted > 0 {
                lg.Info(fmt.Sprintf("Clear expired equipment block: %d equipment_status record(s) deleted", numDeleted))
            }
        }
    })
}
`
Although we do not plan to stop checking for unblocking, it let the task automatically be cancelled by its context. So I agree that it's worth to re-work the module.

Reply:
Ah, I see you already created a commit for it. Cool

Reply:
Also I was telling about unified function for running submerging periodically, not only an equipment function. Just any function that have a particular signature. I guess for now we can skip this feature